### DISCUSSION TITLE
How do I add ContentParts to tabs within the admin editor, for a ContentType, as I can with ContentFields?

### DISCUSSION THREAD

**Discussion by ApacheTech** [OrchardCMS/OrchardCore#15657] [Q&A] [answered]
### TL;DR:

You can move a field to a tab by adding `"place": "Parts#Tab Name"` to the placement for the admin editor for the type. How do you do the same, for parts/named parts?

Also, how do I rename/remove the "Content" tab that gets added when I create a named tab?

### Background:

I'm just starting to learn OrchardCore, as we are migrating an API Docs site over from Umbraco. Within Umbraco, the idea of Admin Tabs for type editors is baked into the design, and I want to replicate the end result. We have some complex types we need to move over. I am **only** working with the visual editors within the default admin theme. My experience is fully OOB. Using CMS as a CMS is supposed to be used.

### Scenario:

Say I have a ContentType called `ApiEndpoint`. Within the type, I have a field called `EndpointUrl`. I can move `EndpointUrl` by editing the placement of the field within the `ApiEndpoint` admin editor. The resulting placement for `TextField_Edit` gets added within the Design > Placements page:

```json
[
  {
    "place": "Parts:0#Details;0",
    "displayType": null,
    "differentiator": "ApiEndpoint-EndpointUrl",
    "alternates": null,
    "wrappers": null,
    "shape": null
  }
]
```

However, when I do this, I get two tabs. "Content", and "Details". I only want one tab, "Details". "Content" is empty, and redundant. It shouldn't be there.

Now, I add a named `BagPart` called `Parameters` to `ApiEndpoint`. It shows up in the "Content" tab. But no combination of adding any kind of `"place": "Parts:0#Parameters;0",` to any placement will make it move away from the "Content" tab that I don't want, and can't rename.

If I add the `place` to the `BagPart_Edit` shape placement, then the button moves to the correct tab, but the title, and description, don't.

I have a number of parts that I would like to separate correctly into tabs. I'm not sure what I'm missing.

**Comment by ApacheTech** [comment, marked as answer]
### Answer

1. Manually create a placement for `ContentPart_Edit`.
2. Add the following:
```json
[
  {
    "place": "Parts#Parameters",
    "displayType": null,
    "differentiator": "ApiEndpoint-Parameters",
    "alternates": null,
    "wrappers": null,
    "shape": null
  }
]
```

This will move the entire part, including the title and description, into the correct tab.

### Bonus:

Renaming tabs:
https://github.com/OrchardCMS/OrchardCore/discussions/13534

**Reply by Piedone** [reply]
Thanks for sharing the solution!

**Reply by ericrrichards** [reply]
Because I am sure that I am going to come back to this in the future, I'm just going to leave this note for future-me
Placements and Shape naming rules are kind of complicated...

## Scenario:
- ContentPart `{part_name}`, which consists of pre-made ContentFields, with no custom DisplayDriver implemented
- ContentType `{content_type}`

## Goal:
- move the `{part_name}` part in the Editor view to a tab named `{tab_name}`

## Solution
"differentiator": "`{content_type}`**-**`{part_name}`"

```
"ContentPart_Edit": [
  {
    "place": "Parts#{tab_name}",
    "differentiator": "{content_type}-{part_name}"
  }
]
```

**Reply by Piedone** [reply]
Would you perhaps like to contribute this as docs? https://docs.orchardcore.net/en/latest/contributing/contributing-documentation/
